geoh5py/groups/property_group.py

```diff
@@ -355,41 +352,52 @@ def verify_data(self, data: Data | UUID | str) -> UUID:
 
         :return: The uuid of the data.
         """
-        data = str2uuid(data)
+        data = self.reference_to_data(data)
 
-        if isinstance(data, Data):
-            if self.parent != data.parent:
-                raise ValueError(
-                    f"Data '{data.name}' parent ({data.parent}) "
-                    f"does not match group parent ({self.parent})."
-                )
+        if self.association != data.association:
+            raise ValueError(
+                f"Data '{data.name}' association ({data.association}) "
+                f"does not match group association ({self.association})."
+            )
 
-        if isinstance(data, (str, UUID)):
-            data_: list = self.parent.get_data(data)
-            # if the data is an unloaded uid
-            if len(data_) == 0 and isinstance(data, UUID):
-                data_temp = self.parent.workspace.load_entity(data, "data", self.parent)
-                data_ = [] if data_temp is None else [data_temp]
-            if len(data_) == 0:
-                raise ValueError(f"Data '{data}' not found in parent {self.parent}")
-            if len(data_) > 1:
+        return data.uid
+
+    @staticmethod
+    def validate_group_type(value: str | GroupTypeEnum) -> GroupTypeEnum:
+        if isinstance(value, str):
+            try:
+                value = GroupTypeEnum(value)
+            except ValueError as error:
                 raise ValueError(
-                    f"Multiple data '{data}' found in parent {self.parent}"
-                )
-            data = data_[0]
+                    f"'Property group type' must be one of "
+                    f"{', '.join(GroupTypeEnum.__members__)}. Provided {value}"
+                ) from error
 
-        if not isinstance(data, Data):
+        if not isinstance(value, GroupTypeEnum):
             raise TypeError(
-                f"Data must be of type Data, UUID or str. Provided {type(data)}"
+                f"'Property group type' must be of type {GroupTypeEnum}, "
+                f"provided {type(value)}"
             )
 
-        if self._association is None:
```

Comment:
This was quite convenient.

Reply:
Yes but not the behaviour we want I would say. Association is required by geoh5, so I don't think we want allow for None

Reply:
ok I get the logic. If geoh5py requieres it, I'll let it there